core/templates/pages/donate-page/donate-page.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for donate page.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { DonatePageComponent } from './donate-page.component';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+describe('Donate page', () => {
+  const siteAnalyticsServiceStub = new SiteAnalyticsService(
+    new WindowRef());
+
+  beforeEach(async() => {
+    TestBed.configureTestingModule({
+      declarations: [DonatePageComponent],
+      providers: [
+        {provide: SiteAnalyticsService, useValue: siteAnalyticsServiceStub},
+        UrlInterpolationService,
+        {
+          provide: WindowDimensionsService,
+          useValue: {
+            isWindowNarrow: () => true
+          }
+        },
+        {
+          provide: WindowRef,
+          useValue: {
+            nativeWindow: {
+              location: {
+                href: ''
+              }
+            }
+          }
+        }
+      ]
+    }).overrideComponent(DonatePageComponent, {
+      set: {
+        // The template is being overwritten because require is being
+        // used on component, which will throw an error on unit tests.
```

Comment:
Hmmm, why?

Reply:
Because the paths work differently when running tests and running the local server (in `templateUrl` property). The path that works on local server will break the unit tests and vice versa. So I've overwritten the `templateUrl` just to work in both environments.

Reply:
Hmmm, ok, but is it not possible to include this file in the karma conf.ts so that it would be recognized here?  By making this change, we can't really ensure that the template is loaded correctly right?

Reply:
I'm not sure about the first question. In this case, the template is not being loaded correctly but I've assumed it's ok because we're only testing its logic. /cc @bansalnitish and @srijanreddy98 do you have any solution in this case here? Thanks!

Reply:
I'm sorry, could you please explain me a bit why do we need such thing? It is not clear from the comment you've added.

Reply:
Changing `template: require(` to` templateUrl: './donate-page.component.html'` after merging with develop should work.

Reply:
@marianazangrossi  Just to clarify, I have added a plugin called `angular2-template-loader`. If you look at the webpack config, this plugin is present before any other HTML loader (it is alongside the `ts-loader` plugin). So this how the compilation happens:

1.  ts-loader goes through the typescript files. It will ignore the templateUrl: './donate-page.component.html' as there is no `require(`.
2. Now, angular2-template-loader goes through the files looking for `@Component({...})`. It sees templateUrl and fetches the file content and copies it into the ts file.
3. The rest of the process continues as it was before.

Since the compilation happens in source directory, it is unaffected by dev/prod. This is because dev/prod is introduced after compilation and used during serving files.

Reply:
Thanks for the help @srijanreddy98